### ISSUE TITLE
Broadsides Does not Detect Mockingboard without Workaround

### ISSUE THREAD

**Issue by GreatHierophant** [AppleWin/AppleWin#608]
The game Broadsides supports the Mockingboard, but it does not detect it as it should through AppleWin v1.27.13.0.  You can get the game to detect the Mockingboard by entering the debug console and setting a breakpoint with BP, it does not matter the value, then reboot the system.

**Comment by tomcw** [comment]
Thanks for reporting this.

I've reproduced with 1.27.13, 1.27.0 and 1.25.0.4 - so not a regression.
(Just set a bp at 0)

**Comment by tomcw** [comment]
Here's the Broadside ISR and Mockingboard detection code:

```
; IRQ loads PC with ($FFFE) = 0xC3FA on Enhanced //e.
; Which eventually vectors to the user ISR...
ISR:
5250:8A                   TXA        
5251:48                   PHA        
5252:98                   TYA        
5253:48                   PHA        
5254:A9 FF                LDA #$FF   
5256:85 06                STA $06    ; set interrupt flag
5258:A9 40                LDA #$40   
525A:8D 0E C4             STA $C40E  ; IER = disable TIMER1 interrupt
525D:68                   PLA        
525E:A8                   TAY        
525F:68                   PLA        
5260:AA                   TAX        
5261:A5 45                LDA A5H/ACC
5263:40                   RTI        ; TIMER1_COUNTER.w is 0xFF47, and time in interrupt is 0xBA cycles
; ...return to INTCX firmware, and then finally RTI to userland after a TOTAL time in interrupt is 0x143 cycles

MockingboardCheck:
5200:A9 00                LDA #$00    
5202:85 06                STA $06     ; clear interrupt flag
5204:A9 00                LDA #$00    
5206:8D 0B C4             STA $C40B   ; ACR = one-shot mode
5209:A9 50                LDA #$50    
520B:8D FE 03             STA IRQL    
520E:A9 52                LDA #$52    
5210:8D FF 03             STA IRQH    
5213:A9 C0                LDA #$C0    
5215:8D 0E C4             STA $C40E   ; IER = enable TIMER1 interrupt
5218:A9 01                LDA #$01    
521A:8D 04 C4             STA $C404   ; TIMER1_LATCH.l = 0x01
521D:A9 00                LDA #$00    
521F:8D 05 C4             STA $C405   ; TIMER1_LATCH.h = 0x00 && TIMER_COUNTER.w = 0x0001
5222:58                   CLI         ; (2)
; AppleWin: in stepping mode, IRQ occurs here
5223:60                   RTS         ; (6)
6BA6:4C 3C 03             JMP $033C	  ; (3)
033C:EA                   NOP         ; (2)
033D:EA                   NOP         ; (2)
033E:EA                   NOP         ; (2)
033F:A5 06                LDA $06
```

**Comment by tomcw** [comment]
Fixed in ceb0416.

**Comment by tomcw** [comment]
NB. Release of AppleWin 1.28.0 containing this fix is [here](https://github.com/AppleWin/AppleWin/releases/tag/v1.28.0.0).

**Comment by GreatHierophant** [comment]
This issue has cropped up again in 1.29.14.0.  Broadsides requires a breakpoint when I load a .woz image of the game or its Apple Speaker sound effects only.

**Comment by tomcw** [comment]
Thanks (and it's not good to have regressed like this).

Regression at 1.29.6.0, due to:
- Fix for 6502 interrupt is delayed by 1 opcode when interrupt occurs on last cycle of opcode.

[developer note] To check if MB was detected or not:
At title screen, after drive LED off: $30F: 0=No MB, 1=MB detected

**Comment by tomcw** [comment]
Fixed by PR #841.

**Comment by tomcw** [comment]
FYI @GreatHierophant : There's a new release of AppleWin 1.29.15.0 containing this fix: [here](https://github.com/AppleWin/AppleWin/releases/tag/v1.29.15.0).
